src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -184,6 +188,20 @@ private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pa
 
         // Used to determine how many threads to use when indexing
         parameters.put(KNNConstants.INDEX_THREAD_QTY, KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY));
+        Object name = parameters.get(NAME);
+        if (name != null && name.equals(METHOD_NSG)) {
+            if (parameters.containsKey(PARAMETERS)) {
+                parameters.remove(PARAMETERS);
+            }
+            /** TODO:
+             * when numIds is too small, NSG graph would core/throw exception
+             * because There are too much invalid entries in the knn graph.
+             */
+            if (pair.docs.length < 256) {
```

Comment:
let move this 256 to a constant. Also do we need to take care of this condition while merging the segments too?

Reply:
LGTM

Reply:
> Also do we need to take care of this condition while merging the segments too

when benchmark tests and other `IT` test case, it will do `forcemerge` to segment 1 to check this condition.